client/extensions/auto-translator/src/widget/translation-dialog/compare-accordion/helpers.ts

```diff
@@ -13,17 +13,54 @@ export const sanitizeHtml = (html: string | Node) => {
 
 export const getPrettyDiffHtml = (diffs: diff_match_patch.Diff[]) => {
   let html = [];
+  const pattern_para = /\n/g;
+  const tempDiv = document.createElement("div");
 
-  for (var x = 0; x < diffs.length; x++) {
+  const getFormattedElements = (text: string, type: "ins" | "del") => {
+    let elements = "";
+    const fragment = document.createRange().createContextualFragment(text);
```

Comment:
We can create a single Range object outside of the loop and reuse it for each createContextualFragment call.

```typescript
const range = document.createRange();

const getFormattedElements = (text: string, type: "ins" | "del"): string => {
  const fragment = range.createContextualFragment(text); 
```

Reply:
I don't think this is necessary since I only want fragment within the function. I would have range variable in memory throughout all iterations which may not be required.